lib/diff.ml

```diff
@@ -149,7 +157,207 @@ let rec type_item ~typing_env ~name ~reference ~current =
+              cur_type_var_map' ) =
+          normalize_type_params [] cur_type_params'
+        in
+        ( normalized_ref_type_params,
+          normalized_cur_type_param :: normalized_cur_type_params',
+          ref_type_var_map,
+          String_map.add cur_type_param_name unique_name cur_type_var_map' )
+    | ref_type_param :: ref_type_params', cur_type_param :: cur_type_params' ->
+        let ref_type_param_name = get_type_param_name ref_type_param in
+        let cur_type_param_name = get_type_param_name cur_type_param in
+        let unique_name = gen_unique_type_var_name false in
+        let normalized_ref_type_param =
+          create_expr (Tvar (Some unique_name))
+            ~level:(get_level ref_type_param) ~scope:(get_scope ref_type_param)
+            ~id:(get_id ref_type_param)
+        in
+        let normalized_cur_type_param =
+          create_expr (Tvar (Some unique_name))
+            ~level:(get_level cur_type_param) ~scope:(get_scope cur_type_param)
+            ~id:(get_id cur_type_param)
+        in
+        let ( normalized_ref_type_params',
+              normalized_cur_type_params',
+              ref_type_var_map',
+              cur_type_var_map' ) =
+          normalize_type_params ref_type_params' cur_type_params'
+        in
+        ( normalized_ref_type_param :: normalized_ref_type_params',
+          normalized_cur_type_param :: normalized_cur_type_params',
+          String_map.add ref_type_param_name unique_name ref_type_var_map',
+          String_map.add cur_type_param_name unique_name cur_type_var_map' )
+  in
+  let ( normalized_ref_type_params,
+        normalized_cur_type_params,
+        ref_type_var_name_map,
+        cur_type_var_name_map ) =
+    normalize_type_params
+      Types.(ref_type_decl.type_params)
+      Types.(cur_type_decl.type_params)
+  in
+  let normalized_ref_type_manifest =
+    normalize_type_manifest ref_type_decl.type_manifest ref_type_var_name_map
+  in
+  let normalized_cur_type_manifest =
+    normalize_type_manifest cur_type_decl.type_manifest cur_type_var_name_map
+  in
+  let normalized_ref_type_kind =
+    normalize_type_kind ref_type_decl.type_kind ref_type_var_name_map
+  in
+  let normalized_cur_type_kind =
+    normalize_type_kind cur_type_decl.type_kind cur_type_var_name_map
+  in
+  ( {
+      ref_type_decl with
+      type_params = normalized_ref_type_params;
+      type_manifest = normalized_ref_type_manifest;
+      type_kind = normalized_ref_type_kind;
+    },
+    {
+      cur_type_decl with
+      type_params = normalized_cur_type_params;
+      type_manifest = normalized_cur_type_manifest;
+      type_kind = normalized_cur_type_kind;
+    } )
+
+and normalize_type_expr tvmap type_expr =
+  match get_desc type_expr with
+  | Tvar None -> type_expr
+  | Tvar (Some name) ->
+      let nname = String_map.find name tvmap in
+      create_expr (Tvar (Some nname)) ~level:(get_level type_expr)
+        ~scope:(get_scope type_expr) ~id:(get_id type_expr)
```

Comment:
Define a `create_expr` function with less arguments, which is just a wrapper to `create_expr`!